### DISCUSSION TITLE
Funding, sponsorships

### DISCUSSION THREAD

**Discussion by pawamoy** [mkdocs/mkdocs#3669] [General]
Previously, the "Sponsor" button linked to oprypin's profile, as he was the main maintainer. With him stepping down, what should we do about funding? He already removed the FUNDING file that creates the sponsor button.

[As suggested by others](https://github.com/mkdocs/mkdocs/issues/892), I think we should use Open Collective, as we already have a setup there: https://opencollective.com/mkdocs (done at the time to [receive money from Bloomberg](https://github.com/mkdocs/mkdocs/discussions/3369)). Today the situation is still a bit unclear regarding maintenance and funding, so Open Collective could at least allow people and companies to sponsor the project, and allow us to split the funds among contributors, or just build up a budget for later expenses/rewards, while we work together towards something clearer.

I'm especially interested in getting @tomchristie's opinion on this, as he as expressed the possibility to [bring MkDocs into the @encode fold](https://github.com/mkdocs/mkdocs-team-space/discussions/14#discussioncomment-9098083).

**Comment by byrnereese** [comment]
Forgive a potentially naive question, but assuming we started taking sponsorships, what would the revenue generated by used for? What costs do we currently have beyond people's time?

**Reply by lovelydinosaur** [reply]
It's primarily people's time and experience we'd be paying for.

I've been working on open source almost entirely full-time over the last 10 years or so, and ended up with all the usual running costs of any other small business. Overheads are relatively low, with salary being the main outgoing. Perhaps I'll manage to get my act together and write-up a comprehensive break-down sometime. This is a really useful prompt, thanks.

**Comment by byrnereese** [comment]
I don't want to complicate things... but if we are going to take donations, we may consider forming a non-profit. 

I think we should absolutely cover any hard costs - e.g. server space, and other recurring costs. 

Compensating people I am in favor of as well, especially on a work-for-hire or contract basis. But if we compensate core team members, then I feel we need to really mature our level of transparency and operations. 

I mean if I am being honest - it would be a dream come true to be able to work on mkdocs full time. But the project can't afford me. :) We would need venture capital money. :-P

**Reply by timvink** [reply]
> I mean if I am being honest - it would be a dream come true to be able to work on mkdocs full time.  But the project can't afford me. :)

Haha same here, I'd love to spend a couple of months at least dedicated to mkdocs. It's a shame really as OS has so many more users behind it than internal software.

**Reply by lovelydinosaur** [reply]
Trading as a limited company while operating in the ethos of a non-profit is a low-touch approach here.
Less red-tape to jump through. That's what [encode.io](https://www.encode.io/) has been doing.

Here's an example of a financial report, when I was on top of things...

https://www.encode.io/reports/december-2021

Nothing like as comprehensive as I'd like it to be, but will give you an broad idea.

The business took a bit of hit over the last couple of years (family reasons).
It's in a decent position currently, but not currently bringing in enough for me to be full time right now.

**Comment by Andre601** [comment]
> I'm especially interested in getting @tomchristie's opinion on this, as he as expressed the possibility to [bring MkDocs into the @encode fold](https://github.com/mkdocs/mkdocs-team-space/discussions/14#discussioncomment-9098083).

Given discussion has been moved now, could you link to the discussion that this one is about? GitHub gives a 404 instead of redirecting to where it would now be (assuming it was moved).

**Reply by lovelydinosaur** [reply]
Yeah that's being a bit annoying in not simply redirecting.

Tracked it down, and @pawamoy is refering [to this comment](https://github.com/mkdocs/mkdocs/discussions/3661#discussioncomment-9098083).

**Comment by squidfunk** [comment]
__Material for MkDocs would really love to sponsor MkDocs__ ❤️

However, it's unclear how? There doesn't seem to be any possibility to sponsor on [OpenCollective](https://opencollective.com/mkdocs) anymore, and there's currently no entity listed as a sponsorable in this organization. Another option that's currently available might be to sponsor @encode, which [seems to take sponsorings](https://github.com/sponsors/encode). This might be an option, since @tomchristie [stepped up as a lead developer again](https://github.com/mkdocs/mkdocs/discussions/3677):

> I'm going to be stepping up as lead maintainer for MkDocs again. [...] One of the pressing issues is that without a proper funding approach we're not going to be able to provide dedicated time towards improving MkDocs.

We'd really like to help out with that. I personally think Material for MkDocs sponsoring MkDocs can be a great sign to motivate other companies and users to follow. It would be great if we can sort out where and how we can initiate the sponsorship ☺️ I'm also not sure, if we want to sponsor a single person, or MkDocs as a whole, so funds can be dispersed among contributors.

**Reply by lovelydinosaur** [reply]
✨ *Fantastic, thanks so much for the support @squidfunk* ✨

I'm still considering exactly how to approach this. To start with, I've setup a `mkdocs` sponsorship page which is currently awaiting approval. Consider it a starting point once it's live.

I'd guess that a significant part of *where should mkdocs be going* is likely to be influenced by *what does Material for MkDocs need from us*. For my part I've got some ideas that are fairly figured out around where this could be going, and some that are still in progress. Expect this to be an ongoing dialog.

🚧 **works ahead** 🚧

 &mdash; Tom ☺️

**Reply by squidfunk** [reply]
That's great news! Setting up a dedicated sponsorable entity for MkDocs on GitHub Sponsors is likely going to be the most low touch approach for collecting the most sponsorships easily. Just know that a downside of GitHub Sponsors compared to OpenCollective is that it's not possible to disperse funding among multiple contributors. We have the same problem, and push funds upstream by directly sponsoring authors of critical MkDocs plugins and extensions, which might also work for you.

> I'd guess that a significant part of where should mkdocs be going is likely to be influenced by what does Material for MkDocs need from us. For my part I've got some ideas that are fairly figured out around where this could be going, and some that are still in progress. Expect this to be an ongoing dialog.

That sounds sensible, albeit I feel it's worth pointing pointing out, that we're not the only part of the MkDocs ecosystem 😅 As we discussed in multiple video calls, I see our effort at Material for MkDocs as part of building one of the best ecosystem for technical documentation, so everything that helps the entire ecosystem to flourish should be a priority, from our stand point.

I will open up some issues addressing those topics in the coming weeks, as well as comment on some that are already open. Additionally, I'll report bugs as we or the community finds them, as Material for MkDocs has quite a high volume of incoming requests, and start opening feature requests for things that we need, but are also useful to other themes and plugins.

> 🚧 works ahead 🚧

💪

**Reply by timvink** [reply]
"I'm still considering exactly how to approach this."

Just came across polars, worth checking out! https://github.com/polarsource/polar

**Reply by squidfunk** [reply]
Jup, Polar might be an option as well. Birk and his team have truly built something amazing! Even more importantly, Polar handles sales tax for you – something that I currently have to do manually, and I can only tell you: it's becoming more and more of a nuissance. I will need to change my approach in the very near future as well, maybe moving away from GiHub Sponsors.

**Reply by Andre601** [reply]
Personally speaking, moving away from Sponsors could allow more people to join in on supporting you Squidfunk.

Tho, downside is, you need more work to make this work auto matically.

But that's a topic for the material repo...

**Comment by pawamoy** [comment]
https://github.com/sponsors/mkdocs

Congrats! :tada: Just started a sponsorship :slightly_smiling_face: 

Just a quick note: it's an organization sponsors profile, but it speaks in the first person ("me", "I"). We understand this "me" is you @tomchristie when looking at "Meet the team" below where you're the only one listed. But maybe it would be better to fully embrace first person and present yourself in the text? So that there's no doubt who is speaking if you ever add other people to "Meet the team" :slightly_smiling_face:

**Reply by squidfunk** [reply]
Jup, I'm also on board! 🚢

**Reply by squidfunk** [reply]
@tomchristie, you might want to add back the [`FUNDING.yml`](https://github.com/mkdocs/mkdocs/pull/3295/files) ✌️
